ban_test.go

```diff
@@ -0,0 +1,148 @@
+	require.NoError(t, err)
+
+	msg = messageResp.Message
+	require.Equal(t, false, msg.Shadowed)
+
+	messageResp, err = c.GetMessage(context.Background(), msg.ID)
+	require.NoError(t, err)
+	require.Equal(t, true, messageResp.Message.Shadowed)
+
+	_, err = c.UnBanUser(context.Background(), userB.ID)
+	require.NoError(t, err)
+
+	msg = &Message{Text: "test message"}
+	messageResp, err = ch.SendMessage(context.Background(), msg, userB.ID)
+	require.NoError(t, err)
+
+	msg = messageResp.Message
+	require.Equal(t, false, msg.Shadowed)
+
+	messageResp, err = c.GetMessage(context.Background(), msg.ID)
+	require.NoError(t, err)
+	require.Equal(t, false, messageResp.Message.Shadowed)
+
+	_, err = ch.UnBanUser(context.Background(), userC.ID)
+	require.NoError(t, err)
+
+	msg = &Message{Text: "test message"}
+	messageResp, err = ch.SendMessage(context.Background(), msg, userC.ID)
+	require.NoError(t, err)
+
+	msg = messageResp.Message
+	require.Equal(t, false, msg.Shadowed)
+
+	messageResp, err = c.GetMessage(context.Background(), msg.ID)
+	require.NoError(t, err)
+	require.Equal(t, false, messageResp.Message.Shadowed)
+}
+
+func TestBanUnbanUser(t *testing.T) {
+	c := initClient(t)
+	target := randomUser(t, c)
+	user := randomUser(t, c)
+
+	_, err := c.BanUser(context.Background(), target.ID, user.ID, BanWithReason("spammer"), BanWithExpiration(60))
+	require.NoError(t, err)
+
+	resp, err := c.QueryBannedUsers(context.Background(), &QueryBannedUsersOptions{
+		QueryOption: &QueryOption{Filter: map[string]interface{}{
+			"user_id": map[string]string{"$eq": target.ID},
+		}},
+	})
+	require.NoError(t, err)
+	require.Equal(t, resp.Bans[0].Reason, "spammer")
+	require.NotZero(t, resp.Bans[0].Expires)
+
+	_, err = c.UnBanUser(context.Background(), target.ID)
+	require.NoError(t, err)
+
+	resp, err = c.QueryBannedUsers(context.Background(), &QueryBannedUsersOptions{
+		QueryOption: &QueryOption{Filter: map[string]interface{}{
+			"user_id": map[string]string{"$eq": target.ID},
+		}},
+	})
+	require.NoError(t, err)
+	require.Empty(t, resp.Bans)
+}
+
+func TestChannelBanUnban(t *testing.T) {
+	c := initClient(t)
+	target := randomUser(t, c)
+	user := randomUser(t, c)
+
+	ch := initChannel(t, c, user.ID, target.ID)
+
+	_, err := ch.BanUser(context.Background(), target.ID, user.ID, BanWithReason("spammer"), BanWithExpiration(60))
+	require.NoError(t, err)
+
+	_, err = ch.UnBanUser(context.Background(), target.ID)
+	require.NoError(t, err)
+
+	resp, err := c.QueryBannedUsers(context.Background(), &QueryBannedUsersOptions{
+		QueryOption: &QueryOption{Filter: map[string]interface{}{
+			"channel_cid": map[string]string{"$eq": ch.CID},
+		}},
+	})
+	require.NoError(t, err)
+	require.Empty(t, resp.Bans)
+}
+
+func ExampleClient_BanUser() {
+	client, _ := NewClient("XXXX", "XXXX")
+
+	// ban a user for 60 minutes from all channel
+	_, _ = client.BanUser(context.Background(), "eviluser", "modUser", BanWithExpiration(60), BanWithReason("Banned for one hour"))
```

Comment:
Shouldn't we make sure there are no errors thrown?

Reply:
Those are examples actually, not test per se. And even if someone runs them from browser, they will fail because there is not a valid client. It's just to give an idea to developer how to use that method.

Reply:
ohh so that's what `ExampleClient` does... I didn't know we had that

Reply:
https://go.dev/blog/examples

Reply:
🤯